src/mapgen_functions.cpp

```diff
@@ -2562,15 +2562,13 @@ void mapgen_generic_house(map *m, oter_id terrain_type, mapgendata dat, int turn
         }
     }
 
-    if ("house_base" == terrain_type.t().id_base ) {
+    if ( "house_base" == terrain_type.t().id_base ) {
         int attempts = 100;
         do {
-            rn = rng(lw + 1, rw - 1);
+            rn = rng( lw + 1, rw - 1 );
             attempts--;
-        } while (m->ter(rn, actual_house_height - 1) != t_floor && attempts);
-        if( m->ter(rn, actual_house_height - 1) == t_floor && attempts && !m->has_furn( rn, actual_house_height - 1 ) ) {
-            m->ter_set(rn, actual_house_height - 1, t_stairs_down);
-        }
+        } while ( m->ter( rn, actual_house_height - 1 ) != t_floor && attempts && !m->has_furn( rn, actual_house_height - 1 ) );
+            m->ter_set( rn, actual_house_height - 1, t_stairs_down );
     }
```

Comment:
These pieces of code aren't equal. The original won't place a stair if no suitable place found after 100 attempts. Yours will. If you meant simplification, you could also rewrite it like this:
```c++
for( int attempts = 0; attempts < 100; attempts++ ) {
    int rn = rng( lw + 1, rw - 1 ); // Local scope is intentional
    if( m->ter( rn, actual_house_height - 1 ) == t_floor && !m->has_furn( rn, actual_house_height - 1 ) ) {
        m->ter_set( rn, actual_house_height - 1, t_stairs_down );
        break; // Job done
    }
}
```